### PR TITLE
thomasgauvin: add limit writes per second

### PR DESCRIPTION
**Pull request by thomasgauvin**

### CODE REVIEW

**Review by Oxyjun — APPROVED** [review]

File: src/content/docs/kv/api/write-key-value-pairs.mdx [hunk 1]
```diff
@@ -136,6 +136,136 @@ await env.NAMESPACE.put(key, value, {
 });
 ```
 
+### Limits to KV writes to the same key
+
+Workers KV has a maximum of 1 write to the same key per second. Writes made to the same key within 1 second will cause rate limiting (`429`) errors to be thrown.
+
+The following example serves only as a demonstration of how multiple writes to the same key may return errors by forcing concurrent writes within a single Worker invocation. This is not a pattern that should be used in production.
```
Comment (thomasgauvin) [inline src/content/docs/kv/api/write-key-value-pairs.mdx]:
```suggestion
The following example serves as a demonstration of how multiple writes to the same key may return errors by forcing concurrent writes within a single Worker invocation. This is not a pattern that should be used in production.
```

File: src/content/docs/kv/api/write-key-value-pairs.mdx [hunk 2]
```diff
@@ -136,6 +136,136 @@ await env.NAMESPACE.put(key, value, {
...
+export default {
+	async fetch(request, env, ctx): Promise<Response> {
+		// Rest of code omitted
+		const key = "common-key";
+		const parallelWritesCount = 20;
+
+		// Helper function to attempt a write to KV with retries
+		const attemptWrite = async (i: number) => {
+			return await retryWithBackoff(async () => {
+				await env.simple_kv_hono_jsx.put(key, `Write attempt #${i}`);
```
Comment (elithrar) [inline src/content/docs/kv/api/write-key-value-pairs.mdx]:
```suggestion
				await env.YOUR_KV_NAMESPACE.put(key, `Write attempt #${i}`);
```

Give the namespace a simple name in examples.

File: src/content/docs/kv/api/write-key-value-pairs.mdx [hunk 3]
```diff
@@ -136,6 +136,136 @@ await env.NAMESPACE.put(key, value, {
...
+export default {
+	async fetch(request, env, ctx): Promise<Response> {
+		// Rest of code omitted
+		const key = "common-key";
+		const parallelWritesCount = 20;
+
+		// Helper function to attempt a write to KV and handle errors
+		const attemptWrite = async (i: number) => {
+			try {
+				await env.simple_kv_hono_jsx.put(key, `Write attempt #${i}`);
```
Comment (thomasgauvin) [inline src/content/docs/kv/api/write-key-value-pairs.mdx]:
```suggestion
				await env. YOUR_KV_NAMESPACE.put(key, `Write attempt #${i}`);
```

File: src/content/docs/kv/api/write-key-value-pairs.mdx (hunk 2 above)
Comment (thomasgauvin) [reply src/content/docs/kv/api/write-key-value-pairs.mdx]: Thanks, oversight, copy pasting between projects

File: src/content/docs/kv/api/write-key-value-pairs.mdx [hunk 4]
```diff
@@ -136,6 +136,136 @@ await env.NAMESPACE.put(key, value, {
 });
 ```
 
+### Limits to KV writes to the same key
+
+Workers KV has a maximum of 1 write to the same key per second. Writes made to the same key within 1 second will cause rate limiting (`429`) errors to be thrown.
```
Comment (thomasgauvin) [inline src/content/docs/kv/api/write-key-value-pairs.mdx:141]:
```suggestion
Workers KV has a maximum of 1 write to the same key per second. Writes made to the same key within 1 second will cause rate limiting (`429`) errors to be thrown.

You should not write more than once per second to the same key. Consider consolidating your writes to a key within a Worker invocation to a single write, or wait at least 1 second between writes.
```

**Pull request merged**